### PR TITLE
chore: new endpoints to trigger WAL-G specific tasks

### PR DESCRIPTION
**Pull request by dragarcia**
New endpoints to trigger shell scripts introduced in https://github.com/supabase/postgres/pull/145.
* GET `/config/walg` - returns current config as `{ raw_contents: <string-of-file-contents> }`
* POST `/config/walg` - sets new config - params: `{ raw_contents: <string-of-file-contents>, restart_services : <bool> }`
---
* POST `/walg/backup`: Trigger a physical backup in the background.
    * Runs this [shell script](https://github.com/supabase/postgres/blob/chore/walg_enhancements/ansible/files/admin_api_scripts/commence_walg_backup.sh).
* POST `/walg/enable`: Enable the sending of WAL files to the S3 bucket via `archive_command`.
    * Runs this [shell script](https://github.com/supabase/postgres/blob/chore/walg_enhancements/ansible/files/admin_api_scripts/enable_walg.sh).
* POST `/walg/disable`: Disable the sending of WAL files to the S3 bucket.
    * Runs this [shell script](https://github.com/supabase/postgres/blob/chore/walg_enhancements/ansible/files/admin_api_scripts/disable_walg.sh).

### CODE REVIEW

**Comment by dragarcia** [comment]
Hold for this - will add more stuff in. Turning back to draft.

**Comment by dragarcia** [comment]
All good again. Added functionality for admin API to be able to edit `/etc/wal-g/config.json` file

File: api/config.go [hunk 1]
```diff
@@ -126,7 +134,7 @@ func (a *API) SetFileContents(w http.ResponseWriter, r *http.Request) error {
 		return sendJSON(w, http.StatusInternalServerError, err.Error())
 	}
 
-	if params.RestartServices {
+	if params.RestartServices && application != "walg" {
```
Comment (dragarcia) [inline api/config.go:137]: there will never be a need to restart anything when the WAL-G configuration changes.

File: README.md [hunk 2]
```diff
@@ -44,6 +44,18 @@ GET `/config/gotrue` - returns current config as `{ raw_contents: <string-of-fil
 
 POST `/config/gotrue` - sets new config - params: `{ raw_contents: <string-of-file-contents>, restart_services : <bool> }`
 
+GET `/config/walg` - returns current config as `{ raw_contents: <string-of-file-contents> }`
+
+POST `/config/walg` - sets new config - params: `{ raw_contents: <string-of-file-contents>, restart_services : <bool> }`
```
Comment (darora) [inline README.md:49]: restart services is ignored, yes?

File: api/walg_backup.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	if err := jsonDecoder.Decode(params); err != nil {
+		return sendJSON(w, http.StatusInternalServerError, err.Error())
+	}
+
+    completedCommand := fmt.Sprintf("%s %d %d", "sudo /root/commence_walg_backup.sh", params.ProjectId, params.BackupId)
+
+	cmd := exec.Command("/bin/sh", "-c", completedCommand)
+	output, err := cmd.Output()
+	if err != nil {
+		return errors.Wrap(err, "failed to execute WAL-G backup")
```
Comment (darora) [inline api/walg_backup.go:31]: log output on failure?

File: api/walg_enable.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"os/exec"
+)
+
+func EnableWALG(w http.ResponseWriter, r *http.Request) error {
+	cmd := exec.Command("/bin/sh", "-c", "sudo /root/enable_walg.sh")
+	output, err := cmd.Output()
+	if err != nil {
+		return errors.Wrap(err, "failed to enable WAL-G")
```
Comment (darora) [inline api/walg_enable.go:14]: output log

File: api/walg_disable.go [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"os/exec"
+)
+
+func DisableWALG(w http.ResponseWriter, r *http.Request) error {
+	cmd := exec.Command("/bin/sh", "-c", "sudo /root/disable_walg.sh")
+	output, err := cmd.Output()
+	if err != nil {
+		return errors.Wrap(err, "failed to disable WAL-G")
```
Comment (darora) [inline api/walg_disable.go:14]: output log

Comment (darora) [reply api/walg_disable.go:14]: (or include output in error returned)

**Review by darora — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (dragarcia) [reply README.md:49]: yes, restart is ignored. Even if somehow a restart=true is sent, nothing will happen as per this [line](https://github.com/supabase/supabase-admin-api/blob/chore/walg_enhancements/api/config.go#L137).

**Pull request merged**